### PR TITLE
Extract interface MapTestCase above MutableMapIterableTestCase.

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

File: unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MapTestCase.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        String value4 = map.merge(5, "5", (v1, v2) -> null);
+        Assert.assertEquals("5", value4);
+        Assert.assertEquals(this.newWithKeysValues(1, "1", 2, "2Two", 4, "4", 5, "5"), map);
+
+        // existing key, remapping function throws exception
+        Assert.assertThrows(IllegalArgumentException.class, () -> map.merge(4, "Four", (v1, v2) -> {
+            throw new IllegalArgumentException();
+        }));
+        Assert.assertEquals(this.newWithKeysValues(1, "1", 2, "2Two", 4, "4", 5, "5"), map);
+    }
```
Comment (motlin) [inline unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MapTestCase.java:193]: While working on MutableMapIterable.merge(), I realized that I had to manually test that the assertions I wrote about our maps also apply to HashMap. So I pulled that test up to here, along with other assertions that apply to any java.util.Map.

File: unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MutableMapIterableTestCase.java
Comment (motlin) [inline unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/MutableMapIterableTestCase.java:1]: The test methods were all pulled up from here, MutableMapIterableTestCase.java

File: unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/HashMapTest.java
Comment (motlin) [inline unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/HashMapTest.java:1]: This is a new test subclass for HashMap, which gives confidence that our assertions apply to all java.util.Maps. We already have similar tests for other JDK types like org.eclipse.collections.test.list.mutable.ArrayListTest and org.eclipse.collections.test.set.mutable.HashSetTest.

**Review by donraab — APPROVED** [review]
LGTM

**Comment by donraab** [comment]
@motlin did you mean to rebase, instead of merge master into this branch?

**Comment by motlin** [comment]
> @motlin did you mean to rebase, instead of merge master into this branch?

That's what I get for trying to update the PR on mobile. It's rebased now.

**Pull request merged**